### PR TITLE
feat: add feature config flag for controlling if MLS is enabled AR-2332

### PR DESCRIPTION
**Pull request by typfel**
<table>
<td>
  <a href="https://wearezeta.atlassian.net/browse/AR-2332" title="AR-2332" target="_blank">TaskAR-2332</a>  [AR] disabling MLS for the team during M4
  </td></table>
  <br />
 

----
#### PR Submission Checklist for internal contributors

- The **PR Title**
  - [ ] conforms to the style of semantic commits messages¹ supported in Wire's Github Workflow²
  - [ ] contains a reference JIRA issue number like `SQPIT-764`
  - [ ] answers the question: _If merged, this PR will: ..._ ³

- The **PR Description**
  - [ ] is free of optional paragraphs and you have filled the relevant parts to the best of your ability
----

# What's new in this PR?

Disable MLS support for all builds except dev builds.

### Dependencies

Needs releases with:

- [ ] https://github.com/wireapp/kalium/pull/945

### Attachments (Optional)

_Attachments like images, videos, etc. (drag and drop in the text box)_

----
#### PR Post Submission Checklist for internal contributors (Optional)

 - [ ] Wire's Github Workflow has automatically linked the PR to a JIRA issue
----
#### PR Post Merge Checklist for internal contributors

 - [ ] If any soft of configuration variable was introduced by this PR, it has been added to the relevant documents and the CI jobs have been updated.
----
##### References
1. https://sparkbox.com/foundry/semantic_commit_messages
1. https://github.com/wireapp/.github#usage
1. E.g. `feat(conversation-list): Sort conversations by most emojis in the title #SQPIT-764`.

### CODE REVIEW

**Comment by AndroidBob** [comment]
Build [1](https://10.10.124.134/job/wire-android-reloaded/job/PR-1018/1/) **failed**.

**Comment by AndroidBob** [comment]
Build [2](https://10.10.124.134/job/wire-android-reloaded/job/PR-1018/2/) **succeeded**.

The build produced the following APK's:
- [com.wire.android-vv0.0.1(4489140)(4489140)-dev-release.apk](https://z-lohika.s3-eu-west-1.amazonaws.com/megazord/android/reloaded/dev/release/com.wire.android-vv0.0.1(4489140)(4489140)-dev-release.apk)

**Comment by AndroidBob** [comment]
Build [3](https://10.10.124.134/job/wire-android-reloaded/job/PR-1018/3/) **succeeded**.

The build produced the following APK's:
- [com.wire.android-vv0.0.1(4489245)(4489245)-dev-release.apk](https://z-lohika.s3-eu-west-1.amazonaws.com/megazord/android/reloaded/dev/release/com.wire.android-vv0.0.1(4489245)(4489245)-dev-release.apk)

File: app/src/main/kotlin/com/wire/android/ui/home/newconversation/newgroup/NewGroupScreen.kt [hunk 1]
```diff
@@ -121,7 +121,7 @@ fun NewGroupScreenContent(
                     }
 
                 }
-                if (mlsEnabled || BuildConfig.PRIVATE_BUILD) {
+                if (mlsEnabled || (BuildConfig.PRIVATE_BUILD && BuildConfig.MLS_SUPPORT_ENABLED)) {
```
Comment (mchenani) [inline app/src/main/kotlin/com/wire/android/ui/home/newconversation/newgroup/NewGroupScreen.kt:124]: I think we can remove the private build here, and  only rely on the MLS_SUPPORT_ENABLED field.

File: default.json [hunk 2]
```diff
@@ -22,6 +22,7 @@
         "logging_enabled": true,
         "safe_logging": false,
         "private_build": true,
+        "mls_support_enabled": true,
```
Comment (mchenani) [inline default.json:25]: don't we need to add this field to other configs/build types?

**Review by mchenani — APPROVED** [review]
LGTM, but I have some questions!

File: app/src/main/kotlin/com/wire/android/ui/home/newconversation/newgroup/NewGroupScreen.kt (hunk 1 above)
Comment (typfel) [reply app/src/main/kotlin/com/wire/android/ui/home/newconversation/newgroup/NewGroupScreen.kt:124]: No we can't without changing the behaviour. 

We are controlling two separate things:
1. Is MLS support enabled
2. Is your user allowed to create MLS conversations 

The PRIVATE_BUILD is in this case by passing the team admin config which white lists users for creating MLS conversations.

File: default.json (hunk 2 above)
Comment (typfel) [reply default.json:25]: No, because I've defined it as `mls_support_enabled: false` at the top level, so all other build flavour will have it disabled.

**Comment by AndroidBob** [comment]
Build [4](https://10.10.124.134/job/wire-android-reloaded/job/PR-1018/4/) **failed**.

**Comment by AndroidBob** [comment]
Build [5](https://10.10.124.134/job/wire-android-reloaded/job/PR-1018/5/) **succeeded**.

The build produced the following APK's:
- [com.wire.android-vv0.0.1(4490375)(4490375)-dev-release.apk](https://z-lohika.s3-eu-west-1.amazonaws.com/megazord/android/reloaded/dev/release/com.wire.android-vv0.0.1(4490375)(4490375)-dev-release.apk)

**Pull request merged**